vita3k/gui/src/controls_dialog.cpp

```diff
@@ -15,45 +15,276 @@
+        host.cfg.keyboard_button_l1 = 0;
+    }
+    if (host.cfg.keyboard_button_r1 < 0 || host.cfg.keyboard_button_r1 > 231) {
+        host.cfg.keyboard_button_r1 = 0;
+    }
+    if (host.cfg.keyboard_button_l2 < 0 || host.cfg.keyboard_button_l2 > 231) {
+        host.cfg.keyboard_button_l2 = 0;
+    }
+    if (host.cfg.keyboard_button_r2 < 0 || host.cfg.keyboard_button_r2 > 231) {
+        host.cfg.keyboard_button_r2 = 0;
+    }
+    if (host.cfg.keyboard_button_l3 < 0 || host.cfg.keyboard_button_l3 > 231) {
+        host.cfg.keyboard_button_l3 = 0;
+    }
+    if (host.cfg.keyboard_button_r3 < 0 || host.cfg.keyboard_button_r3 > 231) {
+        host.cfg.keyboard_button_r3 = 0;
+    }
+    if (host.cfg.keyboard_button_triangle < 0 || host.cfg.keyboard_button_triangle > 231) {
+        host.cfg.keyboard_button_triangle = 0;
+    }
+    if (host.cfg.keyboard_button_circle < 0 || host.cfg.keyboard_button_circle > 231) {
+        host.cfg.keyboard_button_circle = 0;
+    }
+    if (host.cfg.keyboard_button_cross < 0 || host.cfg.keyboard_button_cross > 231) {
+        host.cfg.keyboard_button_cross = 0;
+    }
+    if (host.cfg.keyboard_button_square < 0 || host.cfg.keyboard_button_square > 231) {
+        host.cfg.keyboard_button_square = 0;
+    }
+    if (host.cfg.keyboard_leftstick_left < 0 || host.cfg.keyboard_leftstick_left > 231) {
+        host.cfg.keyboard_leftstick_left = 0;
+    }
+    if (host.cfg.keyboard_leftstick_right < 0 || host.cfg.keyboard_leftstick_right > 231) {
+        host.cfg.keyboard_leftstick_right = 0;
+    }
+    if (host.cfg.keyboard_leftstick_up < 0 || host.cfg.keyboard_leftstick_up > 231) {
+        host.cfg.keyboard_leftstick_up = 0;
+    }
+    if (host.cfg.keyboard_leftstick_down < 0 || host.cfg.keyboard_leftstick_down > 231) {
+        host.cfg.keyboard_leftstick_down = 0;
+    }
+    if (host.cfg.keyboard_rightstick_left < 0 || host.cfg.keyboard_rightstick_left > 231) {
+        host.cfg.keyboard_rightstick_left = 0;
+    }
+    if (host.cfg.keyboard_rightstick_right < 0 || host.cfg.keyboard_rightstick_right > 231) {
+        host.cfg.keyboard_rightstick_right = 0;
+    }
+    if (host.cfg.keyboard_rightstick_up < 0 || host.cfg.keyboard_rightstick_up > 231) {
+        host.cfg.keyboard_rightstick_up = 0;
+    }
+    if (host.cfg.keyboard_rightstick_down < 0 || host.cfg.keyboard_rightstick_down > 231) {
+        host.cfg.keyboard_rightstick_down = 0;
+    }
+    if (host.cfg.keyboard_button_psbutton < 0 || host.cfg.keyboard_button_psbutton > 231) {
+        host.cfg.keyboard_button_psbutton = 0;
+    }
+}
+
+static void remapper_button(GuiState &gui, HostState &host, int *button) {
+    if (ImGui::Button(SDL_key_to_string[*button])) {
+        gui.oldCapturedKey = *button;
+        gui.isCapturingKeys = true;
+        while (gui.isCapturingKeys == true) {
+            handle_events(host, gui);
+            *button = gui.capturedKey;
+            check_for_sdlrange(gui, host);
```

Comment:
We can replace this with:
```suggestion
            if (*button < 0 || *button > 231)
                *button = 0;
```